analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -59,6 +60,71 @@ public CSharpSymbolUsageCollector(Compilation compilation, IEnumerable<ISymbol>
             knownSymbolNames = knownSymbols.Select(GetName).ToHashSet();
         }
 
+        // TupleExpression "(a, b) = qix"
+        // ParenthesizedVariableDesignation "var (a, b) = quix" inside a DeclarationExpression
+        public override void VisitAssignmentExpression(AssignmentExpressionSyntax node)
+        {
+            var assignmentLeft = node.Left;
+            Func<int> getLeftTupleCount = null;
+            // in a newer Roslyn API, we could check the Operation to be DeconstructionAssignment
+            if (TupleExpressionSyntaxWrapper.IsInstance(assignmentLeft))
+            {
+                getLeftTupleCount = () => ((TupleExpressionSyntaxWrapper)assignmentLeft).Arguments.Count;
+            }
+            else if (DeclarationExpressionSyntaxWrapper.IsInstance(assignmentLeft)
+                && ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(((DeclarationExpressionSyntaxWrapper)assignmentLeft).Designation))
+            {
+                getLeftTupleCount = () =>
+                {
+                    var declarationExpression = (ParenthesizedVariableDesignationSyntaxWrapper)((DeclarationExpressionSyntaxWrapper)assignmentLeft).Designation;
+                    return declarationExpression.Variables.Count;
+                };
+            }
+
+            if (getLeftTupleCount != null)
+            {
+                var assignmentRight = node.Right;
+                var semanticModel = GetSemanticModel(assignmentRight);
+                var namedTypeSymbol = GetNamedTypeSymbol(semanticModel.GetSymbolInfo(assignmentRight).Symbol);
+                if (namedTypeSymbol != null)
+                {
+                    var deconstructors = namedTypeSymbol.GetMembers("Deconstruct");
+                    if (deconstructors.Length == 1)
+                    {
+                        UsedSymbols.Add(deconstructors.First());
+                    }
+                    else if (deconstructors.Length > 1
+                        && FindDeconstructor(deconstructors, getLeftTupleCount()) is ISymbol deconstructor)
+                    {
+                        UsedSymbols.Add(deconstructor);
+                    }
+                }
+            }
+
+            base.VisitAssignmentExpression(node);
+
+            static INamedTypeSymbol GetNamedTypeSymbol(ISymbol symbol)
+            {
+                if (symbol != null)
+                {
+                    if (symbol.GetSymbolType() is INamedTypeSymbol namedType)
+                    {
+                        return namedType;
+                    }
+                    else if (symbol is IMethodSymbol methodSymbol)
```

Comment:
I like this `methodSymbol` logic very much :) Can we put it inside `GetSymbolType` itself?

I went through it's usages and it's harmless in most cases and it seems useful in the rest, like here https://github.com/SonarSource/sonar-dotnet/blob/68eee4117874b14e22850725d2cbf7867e82d64b/analyzers/src/SonarAnalyzer.CSharp/Rules/ConsumeValueTaskCorrectly.cs#L115

After that, we can just remove this local function.